### PR TITLE
[Query]: Adds ability to choose global vs local/focused statistics for FullTextScore

### PR DESCRIPTION
**Pull request by aayush3011**
# Description

Why?

Cosmos DB's implementation of FullTextScore computes BM25 statistics (term frequency, inverse document frequency, and document length) across all documents in the container, including all physical and logical partitions.

While this provides a valid and comprehensive representation of statistics for the entire dataset, it introduces challenges for several common use cases:

- Multi-tenant scenarios: Tenants often operate in very different domains, which can significantly change the distribution and importance of keywords. Using global statistics leads to distorted relevance rankings for individual tenants.
- Large containers with many partitions: Computing statistics across hundreds or thousands of physical partitions can be time-consuming and expensive. Customers may prefer statistics derived from only a subset of partitions to improve performance and reduce RU consumption.

What?

This PR extends the flexibility of BM25 scoring so that developers can choose between:

- Global (default): FullTextScore computes BM25 statistics across all documents in the container, regardless of any partition key filters. This is the existing behavior.
- Local: When a query includes a partition key filter, BM25 statistics are computed only over the subset of documents within the specified partition key values. Scores and ranking reflect relevance within that partition-specific slice of data.

How?

A new CosmosFullTextScoreScope enum and setFullTextScoreScope() method are added to CosmosQueryRequestOptions:

```java
CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
options.setFullTextScoreScope(CosmosFullTextScoreScope.LOCAL);
options.setPartitionKey(new PartitionKey(tenantId));
   
container.queryItems(
   "SELECT TOP 10 * FROM c WHERE c.tenantId = @tenantId ORDER BY RANK FullTextScore(c.text, 'keywords')",
   options,
   Document.class
);
```

When CosmosFullTextScoreScope.LOCAL is set, the hybrid search aggregator uses only the query's target partition ranges (instead of all ranges) when executing the global statistics query. This is a client-side only change — no new HTTP headers are sent to the backend.

Bug Fixes (discovered during development)

1. **NullPointerException in DocumentQueryExecutionContextFactory.tryCacheQueryPlan:** 
When executing hybrid search queries with a partition key filter, getQueryInfo() returned null  (hybrid search queries use hybridSearchQueryInfo instead), causing a NPE in query plan caching. Added a null guard.
2. **Race condition in HybridSearchDocumentQueryExecutionContext.getComponentQueryResults:** 
The documentProducers field in ParallelDocumentQueryExecutionContextBase is shared mutable state that was reused across multiple logical operations (global statistics query, component queries). When component queries ran via flatMap (concurrent), or when the global statistics Mono was re-subscribed after component queries had reassigned the field, it caused ConcurrentModificationException and IllegalArgumentException: retries must not be negative. Fixed by introducing a createProducers() helper that wraps super.initialize(), captures the produced list into a local variable, and clears the shared field, ensuring each logical operation (global stats, each component query) gets its own isolated producer list. Additionally changed flatMap to concatMap for component queries to serialize initialization of the parent class's shared metrics state.

Testing

   - Tests validate:
    - GLOBAL scope (default) cross-partition returns all matching results
    - Explicit GLOBAL matches default behavior
    - LOCAL scope + pk="2" returns only pk="2" results
    - LOCAL scope + pk="1" returns only pk="1" results
    - RRF queries work with both LOCAL and GLOBAL scopes

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java [hunk 1]
```diff
@@ -141,8 +142,15 @@ private void initialize(
         this.hybridSearchSchedulingStopwatch.start();
 
         if (hybridSearchQueryInfo.getRequiresGlobalStatistics()) {
+            // When FullTextScoreScope is GLOBAL (default), use allFeedRanges for statistics.
+            // When LOCAL, use only targetFeedRanges for scoped statistics.
+            List<FeedRangeEpkImpl> statisticsTargetRanges =
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java:152]: just a QQ:
-  does LOCAL require user to define a partitionKey? do we need any validation? 
- instead of defining the partitionKey in the request options, what if the partitionKey is defined in the sql query? do we have tests to verify that? 
- do we have tests to verify HPK cases if this is a valid case? 
- do we have tests to verify query + feedRange if this is a valid case?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosFullTextScoreScope.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+/**
+ * Specifies the scope for computing BM25 statistics used by FullTextScore in hybrid search queries.
+ */
+public enum CosmosFullTextScoreScope {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosFullTextScoreScope.java:9]: enum here is fixed? no changes in future?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosQueryRequestOptions.java [hunk 3]
```diff
@@ -532,6 +532,38 @@ public CosmosQueryRequestOptions setQueryAdviceEnabled(boolean queryAdviceEnable
...
     }
 
+    /**
+     * Gets the scope for computing BM25 statistics used by FullTextScore in hybrid search queries.
+     *
+     * @return the scope for computing BM25 statistics. Defaults to {@link CosmosFullTextScoreScope#GLOBAL}.
+     */
+    public CosmosFullTextScoreScope getFullTextScoreScope() {
+        CosmosFullTextScoreScope scope = this.actualRequestOptions.getFullTextScoreScope();
+        return scope != null ? scope : CosmosFullTextScoreScope.GLOBAL;
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosQueryRequestOptions.java]: maybe let this.actualRequestOptions to decide the default value, and here we just return this.actualRequestOptions.getFullTextScoreScope()? else it could cause an inconsistent behavior between the public options vs internal implementation options

**Comment by xinlian12** [comment]
## Deep Review Summary

**PR Intent:** Adds `CosmosFullTextScoreScope` enum (GLOBAL/LOCAL) for controlling BM25 statistics scope in hybrid search queries. Also fixes NPE in query plan caching and race condition (`flatMap` to `concatMap`) in hybrid search execution context.

**Overall Assessment:** The feature is well-designed and matches the .NET SDK's equivalent implementation. The bug fixes are correct. Main concerns are around the null-default inconsistency between public/Impl API layers, missing validation for the LOCAL-without-partition-key edge case, and test coverage gaps.

**Existing Comments:** Only a Copilot summary review (0 inline comments). No overlap with findings below.

| Severity | Count |
|----------|-------|
| :yellow_circle: Recommendation | 6 |
| :green_circle: Suggestion | 3 |
| :speech_balloon: Observation | 2 |

**Top findings:**
1. Inconsistent null-default between public (`GLOBAL`) and Impl (`null`) getters creates a latent correctness trap
2. LOCAL scope without partition key silently degenerates to GLOBAL with no warning
3. `@Ignore` annotation commented out instead of properly removed

---
<sub>:warning: AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosQueryRequestOptionsImpl.java [hunk 4]
```diff
@@ -408,4 +411,24 @@ public String getCollectionRid() {
...
         this.collectionRid = collectionRid;
     }
+
+    /**
+     * Gets the scope for computing BM25 statistics used by FullTextScore in hybrid search queries.
+     *
+     * @return the full text score scope, or null if not set (defaults to GLOBAL behavior).
+     */
+    public CosmosFullTextScoreScope getFullTextScoreScope() {
+        return this.fullTextScoreScope;
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosQueryRequestOptionsImpl.java]: **🟡 Recommendation** · Reliability: Inconsistent Contract

**Inconsistent null-default between public and Impl getters**

The public `CosmosQueryRequestOptions.getFullTextScoreScope()` maps `null` to `GLOBAL`. This Impl getter returns raw `null`. The current usage site calls the public getter (safe), but any future internal code checking `getFullTextScoreScope() == GLOBAL` via the Impl class would get `false` for the default case — a latent correctness trap.

**Suggestion:** Make the Impl getter consistent:
``java
return this.fullTextScoreScope != null ? this.fullTextScoreScope : CosmosFullTextScoreScope.GLOBAL;
```n
---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java [hunk 5]
```diff
@@ -141,8 +142,15 @@ private void initialize(
...
+                this.cosmosQueryRequestOptions.getFullTextScoreScope() == CosmosFullTextScoreScope.LOCAL
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java]: **🟡 Recommendation** · Reliability: Silent Misconfiguration

**LOCAL scope without partition key silently degenerates to GLOBAL**

When `CosmosFullTextScoreScope.LOCAL` is set but no partition key is specified, `targetFeedRanges` equals `allFeedRanges` (all partitions are targeted). The statistics query silently covers all partitions — identical to GLOBAL. In multi-tenant scenarios (the primary use case for LOCAL), accidentally omitting the partition key defeats the entire purpose of the feature.

**Suggestion:** Add a warning log when LOCAL scope is set but targetFeedRanges equals allFeedRanges:
``java
if (scope == LOCAL && targetFeedRanges.size() == allFeedRanges.size()) {
    logger.warn("LOCAL fullTextScoreScope set but no partition key filter specified; statistics computed across all partitions");
}
```n
---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java [hunk 6]
```diff
@@ -49,7 +51,7 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.fail;
 
-@Ignore("TODO: Ignore these test cases until the public emulator is released.")
+//@Ignore("TODO: Ignore these test cases until the public emulator is released.")
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java]: **🟡 Recommendation** · Maintainability: Ambiguous Change

**@Ignore annotation commented out instead of removed**

The `@Ignore` had a TODO about waiting for the public emulator. Commenting it out (`//@Ignore(...)`) rather than removing it leaves ambiguity — is the emulator now available? Will CI break in environments without it?

**Suggestion:** If the emulator is now available, remove the annotation entirely. If not, keep it and explain the CI strategy.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java [hunk 7]
```diff
@@ -131,7 +136,7 @@ public void hybridQueryTest() {
             .flatMap(feedResponse -> Flux.fromIterable(feedResponse.getResults()))
             .collectList().block();
         assertThat(resultDocs).hasSize(15);
-        validateResults(Arrays.asList("51", "49", "24", "61", "54", "22", "2", "25", "75", "77", "57", "76", "66", "80", "85"), resultDocs);
+        validateResults(Arrays.asList("61", "54", "51", "49", "24", "2", "57", "22", "75", "25", "77", "76", "66", "80", "85"), resultDocs);
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java:169]: **🟡 Recommendation** · Maintainability: Unverifiable Change

**Expected result orderings changed without explanation**

Seven existing test assertions had their expected result orderings changed. The partition key was changed from `/id` to `/pk`, which changes the physical partition layout and consequently BM25 statistics and scoring order. Without explanation, reviewers cannot verify the new expected values are correct.

**Suggestion:** Add a comment explaining why the expected ordering changed (e.g., partition key change redistributes documents, altering BM25 statistics).

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java [hunk 8]
```diff
@@ -254,11 +259,106 @@ public void hybridQueryWeightedRRFTest(){
...
+            .flatMap(feedResponse -> Flux.fromIterable(feedResponse.getResults()))
+            .collectList().block();
+        assertThat(localPk1ResultDocs).isNotNull();
+        assertThat(localPk1ResultDocs).hasSize(1);
+        assertThat(localPk1ResultDocs.get(0).getId()).isEqualTo("2");
+        assertThat(localPk1ResultDocs.get(0).getPk()).isEqualTo("1");
+    }
+
+    @Test(groups = {"query", "split"}, timeOut = TIMEOUT)
+    public void hybridQueryRRFWithLocalStatisticsTest() {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java:360]: **🟡 Recommendation** · Testing: Missing Coverage

**No test for LOCAL scope without partition key**

Tests cover LOCAL + partition key, but not the edge case where LOCAL is set without a partition key. Given the silent degeneration to GLOBAL behavior (finding above), this is the most likely user error scenario and should be tested to verify it degenerates gracefully.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java [hunk 9]
```diff
@@ -389,7 +397,10 @@ private static Mono<List<List<ScoreTuple>>> retrieveComponentScores(Mono<List<Hy
     }
 
     private Flux<Document> getComponentQueryResults(List<FeedRangeEpkImpl> targetFeedRanges, int initialPageSize, DocumentCollection collection, Flux<QueryInfo> rewrittenQueryInfos) {
-        return rewrittenQueryInfos.flatMap(queryInfo -> {
+        // Use concatMap to serialize component query initialization. The parent class has shared mutable
+        // state (documentProducers, metrics trackers) that is not thread-safe for concurrent access.
+        // Each component query still executes its partition queries in parallel via the inner flatMap.
+        return rewrittenQueryInfos.concatMap(queryInfo -> {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java:433]: **🟡 Recommendation** · Architecture: Technical Debt

**concatMap is a band-aid over non-thread-safe shared state**

The `flatMap` to `concatMap` fix correctly prevents `ConcurrentModificationException` by serializing component query initialization. However, the root cause — `ParallelDocumentQueryExecutionContextBase.documentProducers` is a shared mutable field reassigned and repopulated across logical operations — remains unaddressed. Any future change re-introducing concurrency will re-expose the race.

**Suggestion:** Add a TODO noting the architectural debt. Consider passing the producer list as a local variable rather than relying on the mutable field.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java [hunk 10]
```diff
@@ -254,11 +259,106 @@ public void hybridQueryWeightedRRFTest(){
...
         assertThat(results).isNotNull();
         validateResults(
-            Arrays.asList("75", "24", "49", "61", "21", "9", "26", "4", "6", "37"),
+            Arrays.asList("75", "24", "49", "55", "61", "21", "9", "26", "37", "57"),
             results
         );
     }
 
+    @Test(groups = {"query", "split"}, timeOut = TIMEOUT)
+    public void hybridQueryWithLocalStatisticsTest() {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java:298]: **🟢 Suggestion** · Testing: Weak Assertion

**LOCAL scope tests do not assert result ordering**

The LOCAL scope tests assert result count and partition key values, but not the order of results. For a scoring/ranking feature, ordering is the primary output. Different score computation (local vs global statistics) should produce different orderings.

**Suggestion:** Add `validateResults(Arrays.asList(...), results)` assertions for LOCAL scope, similar to the existing GLOBAL tests.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java [hunk 11]
```diff
@@ -262,7 +262,8 @@ synchronized private static void tryCacheQueryPlan(
         SqlQuerySpec query,
         PartitionedQueryExecutionInfo partitionedQueryExecutionInfo,
         Map<String, PartitionedQueryExecutionInfo> queryPlanCache) {
-        if (canCacheQuery(partitionedQueryExecutionInfo.getQueryInfo()) && !queryPlanCache.containsKey(query.getQueryText())) {
+        QueryInfo queryInfo = partitionedQueryExecutionInfo.getQueryInfo();
+        if (queryInfo != null && canCacheQuery(queryInfo) && !queryPlanCache.containsKey(query.getQueryText())) {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java]: **🟢 Suggestion** · Maintainability: Debugging Aid

**NPE fix could benefit from trace logging**

The null guard silently skips caching for hybrid search queries (which use `hybridSearchQueryInfo` instead of `queryInfo`). A trace log would help diagnose query plan cache miss rates.

**Suggestion:**
``java
if (queryInfo == null) {
    logger.trace("Skipping query plan caching: queryInfo is null (likely a hybrid search query)");
    return;
}
```n
---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java (hunk 10 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java:298]: **🟢 Suggestion** · Cross-SDK: Test Coverage Gap

**Test coverage gap relative to .NET SDK**

The .NET SDK's equivalent implementation includes tests that validate score values differ between LOCAL and GLOBAL scope (not just result sets). The Java tests only verify result counts and partition key membership.

**Suggestion:** Consider adding a test that verifies LOCAL and GLOBAL produce different score rankings for the same query when data distribution differs across partitions.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosFullTextScoreScope.java (hunk 2 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosFullTextScoreScope.java:9]: **💬 Observation** · Design: Enum Pattern

**Enum pattern is consistent with codebase**

`CosmosFullTextScoreScope` uses a plain Java `enum`, consistent with `CosmosVectorDataType`, `IndexingMode`, and 15 other enums in the Cosmos data plane `models` package (zero uses of `ExpandableStringEnum` in the data plane). Since this enum is client-side only (never serialized over the wire), omitting `@JsonValue`/`toString()` is acceptable. Per Azure SDK guidelines, plain enums are allowed for input-only types (exception 2).

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java (hunk 9 above)
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java:433]: **💬 Observation** · Performance: Trade-off

**concatMap serializes previously-parallel component query initialization**

For queries with multiple FullTextScore components (e.g., `RRF(FullTextScore(...), FullTextScore(...))`), component query initialization is now sequential. Each component query still executes its per-partition queries in parallel internally. The latency impact depends on the number of components — typically 2-3, so the impact should be minor.

---
<sub>⚠️ AI-generated review — may be incorrect. Agree? → resolve the conversation. Disagree? → reply with your reasoning.</sub>

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosQueryRequestOptions.java (hunk 3 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosQueryRequestOptions.java]: Updated the code

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosQueryRequestOptionsImpl.java (hunk 4 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosQueryRequestOptionsImpl.java]: Updated the code

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java (hunk 5 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java]: Updated the code

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java (hunk 6 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java]: Updated the code

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java (hunk 8 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java:360]: test added

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java (hunk 11 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java]: Updated the code

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java (hunk 10 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java:298]: Added.

Comment (aayush3011) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java:298]: The Java SDK doesn't expose raw BM25 scores in the result set — the ranking order is the observable manifestation. Our tests validate that LOCAL with pk="1" returns different (subset) results than GLOBAL, confirming the scoped statistics take effect.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java (hunk 1 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java:152]: LOCAL doesn't strictly require a partition key — without one, it gracefully degenerates to GLOBAL (all targetRanges == allRanges), and we now log a warning for this. SQL WHERE clause PK filters also narrow targetRanges via the query plan, so LOCAL works regardless of whether PK is set via setPartitionKey() or in the query itself. 

HPK and feedRange are valid scenarios, will add tests for them.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosFullTextScoreScope.java (hunk 2 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosFullTextScoreScope.java:9]: Yes, aligned with the .NET SDK which is already shipped with only Global and Local. No additional values are planned.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java (hunk 9 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java:433]: Added a TODO, will follow up with a PR.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java (hunk 1 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java:152]: Added the tests
